cpp/include/cudf/io/codec.hpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file codec.hpp
+ * @brief cuDF-IO compression and decompression API definitions
+ */
+
+#pragma once
+
+#include <cudf/io/types.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+#include <cstdint>
+#include <vector>
+
+namespace CUDF_EXPORT cudf {
+namespace io {
+/**
+ * @addtogroup io_codec
+ * @{
+ * @file
+ */
+
+/**
+ * @brief Status of a compression/decompression operation.
+ */
+enum class codec_status : uint8_t {
+  SUCCESS,          ///< Successful, output is valid
+  FAILURE,          ///< Failed, output is invalid (e.g. input is unsupported in some way)
+  SKIPPED,          ///< Operation skipped (if conversion, uncompressed data can be used)
+  OUTPUT_OVERFLOW,  ///< Output buffer is too small; operation can succeed with larger output
+};
+
+/**
+ * @brief Summary of a compression/decompression operation.
+ */
+struct codec_exec_result {
```

Comment:
Very optional
```suggestion
struct codec_execution_result {
```

Reply:
For the record, I don't quite agree with the logic here. The codec_status is also status of the execution. Why would we have different naming convention for the two.

That said, this is not too relevant and we can move on.

On the exec vs execution front, exec is already used in nvbench and thrust, so there's an argument to be made that it's more consistent 🤷

Reply:
It's fine, I'd just would like us to acknowledge that this is mostly personal preference we're discussing.

Regardless, I'm not too concerned. It's not like we named it something tragic like `xyz_info` :D

Reply:
Indeed `codec_exec_info` seems better to me 😂